### ISSUE TITLE
After upgrade to 0.6.61 version, variable renaming is not working

### ISSUE THREAD

**Issue by riosdavi** [Microsoft/vscode-go#968]
Provided the following code:
```go
01. func (cluster *Cluster) Update() {
02.	logger.Debug("--- Update begin")
03.	for _, device := range cluster.Devices {
04.
05.		objects, err := getObjectsByDeviceID(device.DeviceId)
06.		if err == nil {
07.			logger.Debug("--- Update count:", len(objects), " - deviceId:", device.DeviceId)
08.			err = updateObjectCount(objects)
09.		}
10.		if err != nil {
11.			logger.Error(err.Error())
12.		}
13.	}
14.	logger.Debug("--- Update end")
15. }
```
Renaming `device` to `soDevice` causes the following refactoring, where line4 (blank line) is replaced by a duplicate of line 5 with the old variable name (`device`), and lines 8, 9 and 10 are replaced by `logger.Debug("Index`:
```go
01. func (cluster *Cluster) Update() {
02.	logger.Debug("--- Update begin")
03.	for _, soDevice := range cluster.Devices {
04.		objects, err := getObjectsByDeviceID(device.DeviceId)
05.		objects, err := getObjectsByDeviceID(soDevice.DeviceId)
06.			logger.Debug("--- Update count:", len(objects), " - deviceId:", device.DeviceId)
07.			logger.Debug("Index
08.			logger.Error(err.Error())
09.		}
10.	}
11.	logger.Debug("--- Update end")
12. }
```
Removing line 4 (empty line) causes the line 5 not being duplicated after the refactor but the result is almost the same.

***Note***: after removing the logging lines (`logger.Debug` and `logger.Error`), renaming worked as expected.

**Comment by ramya-rao-a** [comment]
@riosdavi We havent changed anything in the rename area in quite a while.

The rename feature works by parsing the diff returned by the `gorename` tool.
The `gorename` tool uses the `diff` tool in your path.

Based on your description, looks like there is mistake with parsing.

Can you change the "--- Update" to just "-- Update" and try again?

My guess is that https://github.com/Microsoft/vscode-go/blob/master/src/diffUtils.ts#L175 is messing up your scenario.

**Comment by riosdavi** [comment]
@ramya-rao-a thanks for your quick reply. Effectively, replacing "---" by "--" solved the issue but I think a better algorithm should be used to detect `diff` output.
I would vote for such a fix.
Thanks!

**Comment by ramya-rao-a** [comment]
@riosdavi Of course, a better logic is needed. I just asked you to to the replacing to confirm that https://github.com/Microsoft/vscode-go/blob/master/src/diffUtils.ts#L175  indeed is the root cause.

Nevertheless, that was put in place as a workaround for a bug in the diff module which has since been fixed, so I can remove that check altogether.

Thanks for reporting the issue and confirming my hypothesis.

Fixed with https://github.com/Microsoft/vscode-go/commit/1adae1f5ca29cab14655e2f5537b75f2084c9d4a

The fix will be out in the next update which should be after about 2 or 3 weeks

**Comment by ramya-rao-a** [comment]
The fix is now available in the latest update (0.6.62)
